scripts/release/asset_publish.py

```diff
@@ -14,20 +14,26 @@
 from subprocess import PIPE, STDOUT, run
 from tempfile import TemporaryDirectory
 from collections import defaultdict
-from typing import List
+from typing import List, Union
 import azureml.assets as assets
 from azureml.assets.model.mlflow_utils import MLFlowModelUtils
 import azureml.assets.util as util
 import yaml
 from azureml.assets.config import PathType
 from azureml.assets.model import ModelDownloadUtils
 from azureml.assets.util import logger
-from azure.ai.ml.entities._load_functions import load_model
+from azure.ai.ml import MLClient, load_component, load_model
+from azure.ai.ml.entities import Component, Environment, Model
+from azure.identity import DefaultAzureCredential
 
-ASSET_ID_TEMPLATE = Template(
-    "azureml://registries/$registry_name/$asset_type/$asset_name/versions/$version")
+
+ASSET_ID_TEMPLATE = Template("azureml://registries/$registry_name/$asset_type/$asset_name/versions/$version")
 TEST_YML = "tests.yml"
 PUBLISH_ORDER = [assets.AssetType.ENVIRONMENT, assets.AssetType.COMPONENT, assets.AssetType.MODEL]
+WORKSPACE_ENV_PATTERN = re.compile(r"^([\w_-]+)(?:\:([0-9.]+)|\@([a-z]+))$")
+REGISTRY_ENV_PATTERN = re.compile(
+    r"^azureml://registries/[\w_-]+/environments/([\w_-]+)/(?:versions/([0-9.]+)|labels/([a-z]+))"
```

Comment:
Simplifying a bit. See https://regex101.com/r/fvDNmy/1
```suggestion
    r"^azureml://registries/.+/environments/(.+)/(?:versions/(.+)|labels/(.+))"
```

Reply:
![image](https://user-images.githubusercontent.com/61145377/222690011-3b46d289-3f14-4939-a2d0-162c7ab77bc1.png)

@elesel I got this comment at top by @skasturi to not have it this broad. Since we are using internally and mostly to get group values (name, version, label) and not using in validation gates/workflows to make sure naming is as per standard, going broader can be okay. Do we plan to make use of regex to validate spec details in future?

Reply:
Both versions and labels can be alphanumeric. I don't see the point of making the regex more strict, especially when we're already checking for a few static elements.

Reply:
updated